src/main.rs

```diff
@@ -19,12 +21,37 @@ struct Args {
     /// The OpenOCD RTT server port
     #[arg(default_value_t = 9090)]
     port: u16,
+
+    /// Dump target for OpenOCD RTT data
+    #[arg(default_value = "")]
+    file: String,
+}
+
+/// Format data string
+fn format_data(data:String) -> std::string::String {
+    let now = Local::now().format("%T%.3f");
+    let formatted_string = format!("{} {}", format!("[{}]", now).bright_black(), data);
+    return formatted_string;
+}
+
+/// Append to file (creates file if it does not exist)
+fn append_file(path: String, data:String) {
+    let mut file = OpenOptions::new()
+        .create(true)
+        .append(true) // append(true) implies write(true)
+        .open(path)
+        .unwrap();
+
+    if let Err(e) = writeln!(file, "{}", data) {
+        eprintln!("Couldn't write to file: {}", e);
+    }
```

Comment:
Some suggestions here:
- I'd suggest to not open the file for every write operation, and instead opening it once the connection is established and closing it when the connection is dropped or the program terminates. Having to open the file for every write operation might add some overhead, especially for frequent writes
- Also here, I would remove the function, since it's being called only once. Moreover, having a function that requires arguments passed by value will force you to use `.clone()` later, adding the cloning overhead
- Consider adding an error message also if the opening of the file fails. Moreover, consider adapting the error message style to the one at line 73